docs/metrics.md

```diff
@@ -6,6 +6,7 @@ You can find information about the source modules ingested below. Remember that
 - [GARD](metrics/gard.md)
 - [ICD10CM](metrics/icd10cm.md)
 - [ICD10WHO](metrics/icd10who.md)
+- [ICD11Foundation](metrics/icd11foundation.md)
```

Comment:
## Error creating metrics (failed component)
### Overview
@matentzn @twhetzel Ugh, this is a major blocker for this work. I don't know why, but after [renaming](https://github.com/monarch-initiative/mondo-ingest/pull/412#discussion_r1485352186) `icd11.owl` to `icd11foundation.owl`, I'm now getting an `INVALID ONTOLOGY FILE ERROR` creating `components/icd11foundation.owl`. I noticed this when running `sh run.sh make ../../docs/metrics/icd11foundation.md`.

### Additional details - cleanup prep
Before running this command for `icd11foundation`, I knew I had to clean up the previous setup / outputs I jhad `icd11`. In `mondo-ingest-odk.yml`, I renamed to `icd11foundation`. I renamed `icd11.yml` to `icd11foundation.yml`, deleted the `metrics/icd11.md` and the JSON that gets created along with that. Then, I ran `update_repo` twice. I then ran the command to make the sources `.md` without trouble, but the metrics `.md` errored because it depends on the component.

### Investigation
#### 1. Seeing if `icd11.owl` errored this time (it did)
Last week I was able to do this whole set up (`update_repo`, `metrics/icd11.md`, etc) under the name `icd11` without error. However now, after I got the error for doing this with the name `icd11foundation`, I did the steps for "Additional details - cleanup prep" again and ran it as I did before with the previous name `icd11`, but I also got an `INVALID ONTOLOGY FILE ERROR`.

#### 2. Running with `-vvv`
I added `-vvv` to the command in `Makefile` and ran `sh run.sh make ../../docs/metrics/icd11foundation.md` and saved the output in: [`icd11foundation metrics -vvv error log.txt`](https://drive.google.com/file/d/17TAq7_8pRxwIZSV-06BPMe4-G1cWOlob/view?usp=drive_link). I then ran `cat LOG_FILE | grep "Stack trace" -A 1` to get primary error messages without the full stacktrace.

<details><summary>List of unique errors</summary>
<p>

1. columnNumber: 1; An invalid XML character (Unicode: 0x3) was found in the element content of the document
2. Attribute not found: IRI (Line 40 column 82)
3. Encountered " <FULLIRI> "<?xml version=\"1.0\"?> "" at line 1, column 1.
4. IRI included an unencoded space: '32' [line 1]
5. Encountered '<?xml version="1.0"?>' at line 1 column 1.  Expected either 'Ontology:' or 'Prefix:' (Line 1 column 1)       
6. IRI included an unencoded space:   [line 1]
7. Found IOException during parsing [line 1, column 2]       
8. File does not contain a binary RDF document       
9. Could not parse JSONLD [line 1, column 2]       
10. Encountered "" at line 1, column 1.
11. LINENO: 1 - Could not find tag separator ':' in line.
12. Encountered " ">" "<?xml version=\"1.0\"?> "" at line 1, column 1.
13. Did not recognise RDF format object RDFa (mimeTypes=application/xhtml+xml, application/html, text/html; ext=xhtml, html)

All of these errors appear only 1 time except for (1), which appeared twice, (4) came up 3 times, and (6) came up twice.

Some of these look like ones I've seen before and may just be warnings / recoverable errors. Others look like they may be the result of other errors. And some of them look more novel / problematic to me.

</p>
</details> 

### Files
[GoogleDrive folder where I uploaded these files](https://drive.google.com/drive/u/0/folders/1uVXddzeOWQuXVHI2dUkj6etsJCK5jslv)
- [`component-download-icd11foundation.owl.owl`](https://drive.google.com/file/d/1J2wRZoLVJsxOQWHAmmhKxN6J7HYu0rkF/view?usp=drive_link)
- [`icd11foundation metrics full log.txt`](https://drive.google.com/drive/u/0/folders/1uVXddzeOWQuXVHI2dUkj6etsJCK5jslv)
- [`icd11foundation metrics -vvv error log.txt`](https://drive.google.com/file/d/17TAq7_8pRxwIZSV-06BPMe4-G1cWOlob/view?usp=drive_link)

### Possible solutions
#### 1. Go back to `icd11` with previous commit
I usually amend my commits. In this case I did not, so I can perhaps revert back if needed if we want to abandon the whole `icd11foundation` renaming. But I'd prefer to keep trying.

#### 2. Formatting
Maybe there's some chance that it doesn't like OWL functional syntax? Perhaps we can (a) first convert to RDF/XML, or (b) pass `--format ofn`?

<details><summary>Reference: $(COMPONENTSDIR)/icd11foundation.owl</summary>
<p>

```make
component-download-icd11foundation.owl: | $(TMPDIR)
	if [ $(MIR) = true ] && [ $(COMP) = true ]; then $(ROBOT) merge -I https://icd11files.blob.core.windows.net/tmp/whofic-2023-04-08.owl.gz \
	annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $(TMPDIR)/$@.owl; fi

$(COMPONENTSDIR)/icd11foundation.owl: component-download-icd11foundation.owl
	if [ $(COMP) = true ]; then if cmp -s $(TMPDIR)/component-download-icd11foundation.owl.owl $(TMPDIR)/component-download-icd11foundation.owl.tmp.owl ; then echo "Component identical."; \
      else echo "Component is different, updating." &&\
		cp $(TMPDIR)/component-download-icd11foundation.owl.owl $(TMPDIR)/component-download-icd11foundation.owl.tmp.owl &&\
		$(ROBOT) annotate -i $(TMPDIR)/component-download-icd11foundation.owl.owl --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@; fi; fi
```

</p>
</details> 

#### 3. Alternative investigation steps
Nico once gave me these special steps which mostly apply to this situation, which I have not tried yet. We should probably document this somewhere.
> Here is my technique on how to deal with failures where we don't have an intelligible error message:
> 1. Create a minimum subset of the file in question (as small as possible) by successively deleting rows
> 2. Try to isolate the rows that cause the problem
> 3. If you can isolate it, it means that the problem is a missing row
> 4. Create a file that works by looking elsewhere (say, mondo)
> 5. Convert it to the serialsation of the file in question
> 6. Compare both files line by line (the isolated subset and the first 100 rows of the working file)

#### 4. Try to determine which errors are truly problematic, and fix
This is my preferred course for now, but I feel at a loss without help from @matentzn @twhetzel or whoever else might be able to help me troubleshoot this.

Reply:
[Trish wrote](https://github.com/monarch-initiative/mondo-ingest/pull/412#issuecomment-1947682580):

---

@joeflack4 I am starting to look through this. Thanks for posting all the information and stack trace! The first thing I see in the trace is `INVALID ONTOLOGY FILE ERROR Could not load a valid ontology from file: component-download-icd11foundation.owl.owl` where ".owl" is repeated in the file name. 

From previous convos, what is the filename for the ontology after it is unzipped by ROBOT? 

I see this example for `convert` (https://robot.obolibrary.org/convert):
```
robot convert --input annotated.obo.gz \
 --output results/annotated.owl
```
so maybe that is adding the additional ".owl" suffix?

---

If the duplicate ".owl" is not a problem, I'd double-check that the file, after being downloaded and unzipped by ROBOT, can still be opened in Protege.

Reply:
@twhetzel Thanks for looking! I moved [your post](https://github.com/monarch-initiative/mondo-ingest/pull/412#issuecomment-1947682580) over here to the this thread and pasted it above.

>  ".owl" is repeated in the file name.
Nico said there is a reason but I don't know what it is yet. But it is normal and expected for `mondo-ingest` (and I think for ODK in general).

>  what is the filename for the ontology after it is unzipped by ROBOT?
It's `tmp/component-download-icd11foundation.owl`! See details for the makefile goal which makes it, as well as how it is used in the `.owl.owl` file.
<details><summary>Details</summary>
<p>

The makefile goal
```make
component-download-icd11foundation.owl: | $(TMPDIR)
	if [ $(MIR) = true ] && [ $(COMP) = true ]; then $(ROBOT) merge -I https://icd11files.blob.core.windows.net/tmp/whofic-2023-04-08.owl.gz \
	annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $(TMPDIR)/$@.owl; fi
```

How it is used in the `.owl.owl`

```make
$(COMPONENTSDIR)/icd11foundation.owl: component-download-icd11foundation.owl
	if [ $(COMP) = true ]; then if cmp -s $(TMPDIR)/component-download-icd11foundation.owl.owl $(TMPDIR)/component-download-icd11foundation.owl.tmp.owl ; then echo "Component identical."; \
      else echo "Component is different, updating." &&\
		cp $(TMPDIR)/component-download-icd11foundation.owl.owl $(TMPDIR)/component-download-icd11foundation.owl.tmp.owl &&\
		$(ROBOT) annotate -i $(TMPDIR)/component-download-icd11foundation.owl.owl --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@; fi; fi
```

</p>
</details>

Reply:
Got to trying to open the file, and it failed. The error was one that looked suspicious to me; the first error that shows up when I run robot in verbose mode:
```
org.xml.sax.SAXParseException; systemId: file:/Users/joeflack4/projects/mondo-ingest/src/ontology/tmp/component-download-icd11foundation.owl.owl; lineNumber: 1288896; columnNumber: 1; An invalid XML character (Unicode: 0x3) was found in the element content of the document.
```

So I guess I can try to replace these characters. I wonder why I didn't get this error last week.

Reply:
Fixed by:
- https://github.com/monarch-initiative/icd11/pull/2

Thanks @twhetzel !